salesagility/SuiteCRM
# added LoginAttempts and checks on the table for IP

## Description



Adds a new module and table (LoginAttempts) in order to track all login attempts to the system, whether they are successful or not, and the IP of the one trying to login. Also takes how many times that IP has failed to login when locking a user out.

## Motivation and Context

Provides auditing for login attempts made to the CRM, which can be expanded to include whitelisting certain users or IPs from the user lockout system or blacklisting specific IPs.

## How To Test This

Set the maxfailedlogins in sugar config and purposefully fail that many times, and it should lock you out as normal. All attempts will be logged in the database.

## Types of changes


### Final checklist
